File: Inclass-Problems/Java/CompareHalves.java
class CompareHalves {
	public static void main(String[] args) {
		System.out.println(solution("badr", "brad"));
		System.out.println(solution("than", "then"));
	}

	public static String solution(String strOne, String strTwo) {
		String strOneHalf = strOne.substring(0, strOne.length() / 2);
		String strTwoHalf = strTwo.substring(0, strOne.length() / 2);

		String answer = strOneHalf.equals(strTwoHalf) 
			? "Halves are the same" 
			: "Halves are not the same";

		return answer;
	}
}

File: Inclass-Problems/Python/compare_halves.py
def compare_halves(word_one: str, word_two: str) -> str:
    return "word halves are equal" if \
        word_one[0:2] == word_two[0:2] else \
        "word halves are not equal"


print(compare_halves('badr', 'brad'))
print(compare_halves('than', 'then'))
